webapp/lib/controller.dart

```diff
@@ -90,6 +96,78 @@ void initUI() {
   );
 }
 
+Map<String, model.NeedsReplyData> getLatestDataForProjects(List<model.NeedsReplyData> updatedData) {
+  Map<String, model.NeedsReplyData> latestProjectData = {};
+  
+  for (var project in projectList) {
+    var data = updatedData.where((data) => data.project == project).toList();
+    
+    if (data.isEmpty) {
+      latestProjectData[project] = null;
+    } else {
+      latestProjectData[project] = data.last;
+    }
+  }
+  return latestProjectData;
+}
+
+bool isProjectStale(model.NeedsReplyData projectData) {
+  var now = new DateTime.now();
+  int lastUpdateTimeDiff =  now.difference(projectData.datetime).inHours;
+
+  if (lastUpdateTimeDiff >= 1) {
+    return true;
+  } else {
+    return false;
+  }
+}
+
+void setupProjectTimer(model.NeedsReplyData projectData, [bool stale = false]) {
+  projectTimers[projectData.project].cancel();
```

Comment:
First time the page loads, `projectTimers[project]` is going to be `null`, so we get an error in the console and the view doesn't get marked as stale even if it needs to be. This fixes it:
```suggestion
  projectTimers[projectData.project]?.cancel();
```